tests/cypress/tests-component/TextTruncate.cy.js

```diff
@@ -0,0 +1,78 @@
+/// <reference types="cypress" />
+
+import TextTruncate from '@/utils/components/TextTruncate.vue'
+
+describe('<TextTruncate />', () => {
+    it('It add a tippy with a simple text slot', () => {
+        const slotContent = 'This is my slot content'
+        cy.mount(TextTruncate, {
+            slots: { default: slotContent },
+            attrs: { style: 'width: 50px' },
+        })
+        cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
+        cy.get('.tippy-box').should('be.visible').contains(slotContent)
+        cy.get('[data-cy="inner-element"]').click('left')
+    })
+
+    it("It don't add a tippy with a complex slot", () => {
+        const slotContent = 'Test<div>This is my slot <b>content</b></div>'
+        cy.mount(TextTruncate, {
+            slots: { default: slotContent },
+            attrs: { style: 'width: 50px' },
+        })
+        cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
+        cy.get('.tippy-box').should('not.exist')
+        cy.get('[data-cy="inner-element"]').click('left')
+    })
+
+    it('It add a tippy with a custom content', () => {
+        const slotContent = 'bla bla bla bla'
+        const textContent = 'Test This is my slot content'
+        cy.mount(TextTruncate, {
+            props: { text: textContent },
+            slots: { default: slotContent },
+            attrs: { style: 'width: 50px' },
+        })
+        cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
+        cy.get('.tippy-box').should('be.visible').contains(textContent)
+        cy.get('[data-cy="inner-element"]').click('left')
+    })
+
+    it('It add a tippy with a custom content and span slot', () => {
+        const slotContent = '<span>My slot content</span>'
+        const textContent = 'My slot content'
+        cy.mount(TextTruncate, {
+            props: { text: textContent },
+            slots: { default: slotContent },
+            attrs: { style: 'width: 50px' },
+        })
+        cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
+        cy.get('.tippy-box').should('be.visible').contains(textContent)
+        cy.get('[data-cy="inner-element"]').click('left')
+    })
+
+    it('It add a tippy after a resize', () => {
+        const slotContent = 'My slot content'
+        cy.mount(TextTruncate, {
+            slots: { default: slotContent },
+        })
+        cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
+        cy.get('.tippy-box').should('not.exist')
+        cy.get('[data-cy="inner-element"]').click('left')
+
+        cy.viewport(50, 480)
+        // Here unfortunately I could not find a way to correctly wait that the
+        // tippy correct got registered after the resize, therefore I've added a wait.
+        // I've tried spying on the initializeTippy method, but somehow the spy was never called.
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(1000)
```

Comment:
@pakb I'm not sure if I want to keep this test, I don't like the 1s wait. Unfortunately I did not find a way to make this test reliable. Without this wait is fails time to time.
I tried to spy on `ResizeObserver` and/or `initializeTippy` but  without success the spy was never called. Maybe you have some better idea ?

Reply:
Here are some code that I tried
```javascript
// Assertions based on the spy

            const slotContent = 'My slot content'
            // const { wrapper, component } = cy.mount(TextTruncate, {
            //     slots: { default: slotContent },
            // })
            cy.mount(TextTruncate, {
                slots: { default: slotContent },
            }).as('mount')
            cy.get('@mount').then((wrapper) => {
                cy.spy(wrapper.component, 'initializeTippy').as('initializeTippy')

                cy.get('@initializeTippy').should('not.have.been.calledOnce')
                cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
                cy.get('[data-cy="outter-element"]').should('not.have.attr', 'aria-describedby')
                cy.get('.tippy-box').should('not.exist')
                cy.get('[data-cy="inner-element"]').click('left')

                cy.viewport(50, 480)
                cy.get('@resizeObserverSpy').should('have.been.calledOnce')
                cy.get('@initializeTippy').should('have.been.calledOnce')
                cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
                cy.get('[data-cy="outter-element"]').should('have.attr', 'aria-describedby')
                cy.get('.tippy-box').should('be.visible').contains(slotContent)
                cy.get('[data-cy="inner-element"]').click('left')

                cy.viewport(300, 480)
                cy.get('@initializeTippy').should('have.been.calledOnce')
                cy.get('[data-cy="inner-element"]').realHover({ position: 'left' })
                cy.get('[data-cy="outter-element"]').should('not.have.attr', 'aria-describedby')
                cy.get('.tippy-box').should('not.exist')
                cy.get('[data-cy="inner-element"]').click('left')
            })

            // Clean-up: restore the original ResizeObserver
            win.ResizeObserver = originalResizeObserver
```